From 6ac61fcdbfc831acd7767bb7a3de8bf5508e9b81 Mon Sep 17 00:00:00 2001
From: Minghuan Lian <Minghuan.Lian@freescale.com>
Date: Tue, 4 Nov 2014 16:28:42 +0800
Subject: [PATCH 297/399] PCI: Layerscape: fix MSI interrupt clearing

MSI_LS1021A_DATA that is used to generate MSI interrupt by PCI
device is a little endian value. It should be converted to big
endian when writing to SCFG_SPIMSICLRCR a big endian register.

Signed-off-by: Minghuan Lian <Minghuan.Lian@freescale.com>
Change-Id: Ie26dd7320f54ee7410d29cca38f4218044549307
Reviewed-on: http://git.am.freescale.net:8181/23098
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Mingkai Hu <Mingkai.Hu@freescale.com>
Reviewed-by: Richard Schmitt <richard.schmitt@freescale.com>
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/pci/host/pci-layerscape.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/pci/host/pci-layerscape.c b/drivers/pci/host/pci-layerscape.c
index e4cd563..8d75caf 100644
--- a/drivers/pci/host/pci-layerscape.c
+++ b/drivers/pci/host/pci-layerscape.c
@@ -90,7 +90,7 @@ static irqreturn_t ls_pcie_msi_irq_handler(int irq, void *data)
 
 	/* clear the interrupt */
 	regmap_write(pcie->scfg, SCFG_SPIMSICLRCR,
-		     MSI_LS1021A_DATA(pcie->index));
+		     cpu_to_be32(MSI_LS1021A_DATA(pcie->index)));
 
 	msi_irq = irq_find_mapping(pp->irq_domain, 0);
 	if (!msi_irq) {
-- 
1.7.5.4

